enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+1. The wrapping controller creates a new AssistedInstallAgent resource to be the k8s-native API for the agent.
+1. CAPBM creates a new BareMetalHost resource, setting the status annotation based on inspection information from the prior step.
+1. The user approves the host for installation. (TODO: how?)
+1. The user or an orchestrator scales up a MachineSet, causing a new Machine resource to be created.
+1. CAPBM binds the Machine to the BareMetalHost, as it does today. It additionally finds the CR representing the agent and uses it to begin installation.
+1. The assisted service initiates installation of the host.
+1. CAPBM updates the status on the BareMetalHost to reflect that it has been provisioned.
+
+#### Day 2 Add Node Virtualmedia Stand-alone
+
+This scenario takes place within a stand-alone bare metal OpenShift cluster.
+
+1. The user creates a BareMetalHost that includes BMC credentials and a label indicating it should be used with assisted installer.
+1. CAPBM gets a URL to the live ISO and adds it to the BareMetalHost, causing it to boot the host.
+1. baremetal-operator uses redfish virtualmedia to boot the live ISO.
+1. The assisted agent starts running on the new hardware and runs through its usual validation and inspection workflow. The wrapping controller creates a new agent resource to be the k8s-native API for the agent.
+1. The user or an orchestrator scales up a MachineSet, resulting in a new Machine being created.
+1. CAPBM does its usual workflow of matching the Machine to an available BareMetalHost. Additionally it uses the agent CR to initiate provisioning of the host.
+1. The assisted service provisions the host.
+
+#### Day 2 Add Node Virtualmedia Multicluster
+
+This scenario takes place from a hub cluster, adding a worker node to a spoke cluster.
+
+1. User creates or modifies a BareMetalAsset on the hub cluster so that its cluster field references the desired ClusterDeployment.
+1. Hive creates a BareMetalHost resource corresponding to the BareMetalAsset and uses it to boot the live ISO.
+1. baremetal-operator uses redfish virtualmedia to boot the live ISO.
+1. Agent running on the host contacts assisted service, resulting in an AssistedInstallAgent resource being created.
+1. Assisted Installer walks the agent through discovery and validation workflows.
+1. When the agent is in a ready state, hive or ACM initiates provisioning, referencing the target cluster's ignition.
+
+#### Create Cluster
+
+This scenario takes place on a hub cluster where hive and possible ACM are
+present. This scenario does not include centralized machine management.
+
+1. User creates BareMetalAssets containing BMC credentials.
+1. User creates AssistedInstallCluster resource describing their desired cluster.
+1. User creates ClusterDeployment describing a new cluster. The Platform section indicates that metal3 should be used to boot hosts with a live ISO. A new field references the AssistedInstallCluster resource.
```

Comment:
@dgoodwin points it out earlier, but this linkage makes me think the new AssistedInstallCluster should just be a new platform within ClusterDeployment. Users shouldn't have to know about metal3.